silencer-plugin/src/main/scala/com/github/ghik/silencer/SuppressingReporter.scala

```diff
@@ -28,10 +31,24 @@ class SuppressingReporter(original: Reporter, globalFilters: List[Regex]) extend
   }
 
   protected def info0(pos: Position, msg: String, severity: Severity, force: Boolean): Unit = {
+    val absolutePath = pos.source.path.replaceAllLiterally("\\", "/")
+    val relativePathOpt =
+      filters.get(SourceRootFilters).flatMap { pathList =>
+        val escapedPaths = pathList.flatMap(_.fold(p => Some(p.replaceAllLiterally("\\", "/")), _ => None))
+        val matchedPath =
+          escapedPaths.collectFirst {
+            case path if absolutePath.contains(path) => Try(absolutePath.substring(path.length)).toOption
```

Comment:
I really don't understand why is this done like this. Why `contains`? Why exception-based logic (`Try`)?

How I would like this to be done:
* `SuppressingReporter` accepts a `sourceRoots: List[File]` as one of the parameters
* Having a `pos.source.file`, we search `sourceRoots` for a root which contains `pos.source.file`. If we find one, we relativize `pos.source.file` against this root. If we don't find one, we leave it absolute.
* Then, we normalize file name separators already in the relativized path.
* Finally, relativized and normalized path is checked against path patterns

Reply:
I've made the changes